File: server/create_admin.py
import os
import bcrypt
from db import db
from dotenv import load_dotenv
from pathlib import Path

load_dotenv(dotenv_path=Path(__file__).parent.parent / '.env')

email    = os.environ.get('ADMIN_EMAIL', 'admin@mindcheck.com')
password = os.environ.get('ADMIN_PASSWORD')
if not password:
    raise EnvironmentError('ADMIN_PASSWORD env var is required')

hashed = bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()

with db() as cur:
    cur.execute(
        "INSERT INTO users (name, email, password, category, role, status, email_verified) "
        "VALUES (%s,%s,%s,'student','admin','approved',TRUE) ON CONFLICT (email) DO NOTHING",
        ('Admin', email, hashed)
    )

print(f'Admin created — email: {email}')

File: server/createAdmin.js
const pool   = require('./db');
const bcrypt = require('bcryptjs');
require('dotenv').config({ path: require('node:path').join(__dirname, '../.env') });

async function main() {
  const email    = 'admin@mindcheck.com';
  const password = 'admin123';
  const hashed   = await bcrypt.hash(password, 10);

  await pool.query(
    `INSERT INTO users (name, email, password, category, role, status, email_verified)
     VALUES ($1, $2, $3, 'student', 'admin', 'approved', TRUE)
     ON CONFLICT (email) DO NOTHING`,
    ['Admin', email, hashed]
  );

  console.log(`Admin created — email: ${email}  password: ${password}`);
  await pool.end();
}

main().catch((err) => { console.error(err); process.exit(1); });
